gigascience/gigadb-website
# Improve accessibility and use of semantic html of search results card

# Pull request for issue: #1374 

This is a pull request for the following functionalities:

* Replace selector `*` that matches all elements by `html, body`. This means that CSS cascading can be used to a greater extent for the defined properties. No pages are visually affected by this change
* Define css variables for color contrast and use them in the search results card styles
* Revamp search results partial for improved semantics and accessibility
  * Card only displayed if there is any content to display
  * wrap in a list: author links, samples, files
  * Improve color contrast between text and background
  * File names are truncated using css rather than programmatically
  * Add accessible label to dataset type icon

## How to test?

* Review that the search results look visually ok
* Run the WAVE Evaluation Tool or other to check that the color contrast between text and background is adequate (note: I opted to leave the text-icon colors untouched as it's more an icon than text content)

## How have functionalities been implemented?

* the _new_results.php file has been updated
* search results styles have been defined in an entirely new module, as I considered this component is too specific to fit a layout definition

## Any issues with implementation?

A sequence of author links is generated from the model directly as a string, so to render a list the string was parsed with php. This could be error prone if sometime the model returns a string with a slightly different shape. I think it would be preferable to have an array of authors and determine the rendering in the php file

## Any changes to automated tests?

The file names that were truncated programmatically are now in full display in the html so some tests that expected a certan value for filenames were updated to accommodate this

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
